os/kernel/signal/sig_dispatch.c

```diff
@@ -440,10 +440,15 @@ int sig_dispatch(pid_t pid, FAR siginfo_t *info)
 
 	stcb = sched_gettcb(pid);
 	if (stcb) {
+
+		if (!sigismember(&stcb->sigprocmask, info->si_signo)) {
```

Comment:
As we discussed on offline, if we should block a signal when we receive it to handle, second one should be pending, should not be passed to another thread.

Reply:
Why don't you commit this change to a new branch for binary separation and IPC?
Signal mechanism modification should be carefully done.